### PR TITLE
Fix qr_jvp (fixes #2863)

### LINKED ISSUE #2863: Gradient of QR fails
There seems to be a major bug in the gradient of the QR decomposition. The forward-mode derivative is accurate for Q but very far off for R. There also seems to be no check in the tests making sure the gradient is accurate.

This reproduces the error:

```
import jax
import jax.numpy as jnp
import numpy as np

x = np.random.randn(3, 3)
dx = np.random.randn(3, 3)

primals, tangents = jax.jvp(jnp.linalg.qr, (x,), (dx,))
q, r = primals
dq, dr = tangents

dt = 1e-6
dq_ = (np.linalg.qr(x + dt * dx)[0] - np.linalg.qr(x)[0]) / dt
dr_ = (np.linalg.qr(x + dt * dx)[1] - np.linalg.qr(x)[1]) / dt

assert jnp.allclose(x, q @ r, atol=1e-5, rtol=1e-5)  # passes
assert jnp.allclose(dq, dq_, atol=1e-5, rtol=1e-5)  # passes
assert jnp.allclose(dx, q @ dr_ + dq_ @ r, atol=1e-5, rtol=1e-5)  # passes
assert jnp.allclose(dr, dr_, atol=1e-5, rtol=1e-5)  # fails
```

### PR DESCRIPTION
**Pull request by j-towns**
The low level qr primitive doesn't outputs a full matrix (not an upper-triangular one) for `r`. Fix this by using the higher level `lax_linalg.qr`.

### CODE REVIEW

**Comment by j-towns** [comment]
Note this isn't ready to be merged yet, see https://github.com/google/jax/issues/2863#issuecomment-620548369.

**Comment by hawkinsp** [comment]
I think I'd rather revert to the first version for the moment (no custom JVP) and instead push the `triu` call into the translation rule, which can be done either by just writing out the logic for it directly or by using `xla.lower_fun`. I'm happy to do that part if you prefer.

**Comment by j-towns** [comment]
> I think I'd rather revert to the first version for the moment (no custom JVP) and instead push the triu call into the translation rule, which can be done either by just writing out the logic for it directly or by using xla.lower_fun. I'm happy to do that part if you prefer.

Would be good if you could. IIUC once triu is pushed down into the primitive this pr will no longer be necessary (i.e. the jvp on master will be correct), except for complex values. If you want I can submit a pr to disable the tests for complex values and raise a NotImlemented, or you can do that.
